File: programmers/level1/로또의 최고 순위와 최저 순위.py
import collections

def solution(lottos, win_nums):
    dic = collections.defaultdict(int)
    for n in win_nums:
        dic[n] = 1
    zero_cnt = 0
    hit = 0
    for n in lottos:
        if n==0: zero_cnt += 1
        elif dic[n] == 1:
            hit += 1
    max_hit = hit + zero_cnt
    min_hit = hit
    max_rank = 7-max_hit
    min_rank = 7-min_hit
    if max_rank > 6: max_rank = 6
    if min_rank > 6: min_rank = 6
    return [max_rank, min_rank]

File: programmers/level1/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        Map<Integer, Integer> lottoMap = new HashMap<>();
        for(int number: win_nums) {
            lottoMap.put(number, 1);
        }
        int hit = 0;
        int zeros = 0;
        for(int predict: lottos) {
            if(lottoMap.containsKey(predict)) {
                hit += 1;
            } else if(predict == 0) zeros++;
        }
        int rank = 7;
        int min_rank = Math.min(6, rank - hit);
        int max_rank = Math.max(Math.min(6, rank - hit - zeros), 1);
        int[] answer = {max_rank, min_rank};
        return answer;
    }
}